geniza/corpus/forms.py

```diff
@@ -0,0 +1,75 @@
+from django import forms
+
+
+class SelectDisabledMixin:
+    """
+    Mixin for :class:`django.forms.RadioSelect` or :class:`django.forms.CheckboxSelect`
+    classes to set an option as disabled. To disable, the widget's choice
+    label option should be passed in as a dictionary with `disabled` set
+    to True::
+
+        {'label': 'option', 'disabled': True}.
+    """
+
+    # copied from mep-django
+
+    # Using a solution at https://djangosnippets.org/snippets/2453/
+    def create_option(
+        self, name, value, label, selected, index, subindex=None, attrs=None
+    ):
+        disabled = None
+
+        if isinstance(label, dict):
+            label, disabled = label["label"], label.get("disabled", False)
+        option_dict = super().create_option(
+            name, value, label, selected, index, subindex=subindex, attrs=attrs
+        )
+        if disabled:
+            option_dict["attrs"].update({"disabled": "disabled"})
+        return option_dict
+
+
+class SelectWithDisabled(SelectDisabledMixin, forms.Select):
+    """
+    Subclass of :class:`django.forms.Select` with option to mark
+    a choice as disabled.
+    """
+
+
+class DocumentSearchForm(forms.Form):
+    query = forms.CharField(
+        label="Keyword or Phrase",
+        required=False,
+        widget=forms.TextInput(
+            attrs={
+                "placeholder": "search by keyword",
+                "aria-label": "Keyword or Phrase",
+            }
+        ),
+    )
+
+    SORT_CHOICES = [
+        ("relevance", "Relevance"),
+        ("input_date", "Input Date (Latest – Earliest)"),
+    ]
+
+    # NOTE these are not set by default!
+    error_css_class = "error"
+    required_css_class = "required"
+
+    sort = forms.ChoiceField(
+        label="Sort by", choices=SORT_CHOICES, required=False, widget=SelectWithDisabled
+    )
+
+    def __init__(self, data=None, *args, **kwargs):
+        """
+        Override to set choices dynamically based on form kwargs.
+        """
+        super().__init__(data=data, *args, **kwargs)
+
+        # if a keyword search term is not present, relevance sort is disabled
+        if data and not data.get("query", None):
+            self.fields["sort"].widget.choices[0] = (
+                self.SORT_CHOICES[0][0],
+                {"label": self.SORT_CHOICES[0][1], "disabled": True},
+            )
```

Comment:
when i tested, relevance was set by default when the page loads, despite there being no query present (although i don't think it's actually sorting by relevance). maybe just doing
```py
if not data or not data.get("query", None):
```
would fix this?

Reply:
I didn't really implement sort yet — think it should be a separate issue. Wasn't sure the best way to communicate that it shouldn't be expected to work  — disable the input, maybe?

The default page load isn't supposed to bring back any records (we display the tag network instead); it seemed simpler to me to leave them there for now. What do you think, is it problematic at all?

Reply:
ah, I'd forgotten about the tag network view. i think leaving the results there is fine, but for the sort dropdown I'd either not render it at all, or add a `hidden` attribute/completely hide it via CSS. we've used the disabled select to mean "this is active, but you can't change it" for sort before (and presumably will again here) so i think until it can function it's better to not show it.

Reply:
alternatively we could just make sure there's no available options other than relevance, and keep it disabled – then only add sort options to the form once they're already implemented